include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,769 @@
+    /* FragmentShadingRateEnumsFeaturesNV *//* VK_NV_fragment_shading_rate_enums */
+    //VkBool32           fragmentShadingRateEnums;
+    //VkBool32           supersampleFragmentShadingRates;
+    //VkBool32           noInvocationFragmentShadingRates;
+
+    // [DO NOT EXPOSE] won't expose, the existing inheritance of state is enough
+    /* InheritedViewportScissorFeaturesNV *//* VK_NV_inherited_viewport_scissor */
+    //VkBool32           inheritedViewportScissor2D;
+
+    // [DO NOT EXPOSE] no idea what real-world beneficial use case would be
+    /* LinearColorAttachmentFeaturesNV *//* VK_NV_linear_color_attachment */
+    //VkBool32           linearColorAttachment;
+    
+    // [TODO]
+    /* MeshShaderFeaturesNV *//* VK_NV_mesh_shader */
+    //VkBool32           taskShader;
+    //VkBool32           meshShader;
+    
+    /* RayTracingMotionBlurFeaturesNV *//* VK_NV_ray_tracing_motion_blur */
+    //VkBool32           rayTracingMotionBlur;
+    //VkBool32           rayTracingMotionBlurPipelineTraceRaysIndirect;
+
+    // [TODO] Expose Soon, Useful
+    /* RepresentativeFragmentTestFeaturesNV *//* VK_NV_representative_fragment_test */
+    //VkBool32           representativeFragmentTest;
+ 
+    // [TODO] requires extra API work to use
+    // GL Hint: in GL/GLES this is NV_scissor_exclusive
+    /* ExclusiveScissorFeaturesNV *//* VK_NV_scissor_exclusive */
+    //VkBool32           exclusiveScissor;
+
+    /* ShaderImageFootprintFeaturesNV *//* VK_NV_shader_image_footprint */
+    //VkBool32           imageFootprint;
+
+    // [ENABLE BY DEFAULT]
+    /* ShaderSMBuiltinsFeaturesNV *//* VK_NV_shader_sm_builtins */
+    //VkBool32           shaderSMBuiltins;
+    
+    // [DO NOT EXPOSE] not implementing or exposing VRS in near or far future
+    /* ShadingRateImageFeaturesNV *//* VK_NV_shading_rate_image */
+    //VkBool32           shadingRateImage;
+    //VkBool32           shadingRateCoarseSampleOrder;
+
+    // [DO NOT EXPOSE] not implementing or exposing VRS in near or far future
+    /* FragmentDensityMapOffsetFeaturesQCOM *//* VK_QCOM_fragment_density_map_offset */
+    //VkBool32           fragmentDensityMapOffset;
+
+    // [DO NOT EXPOSE] This extension is only intended for use in specific embedded environments with known implementation details, and is therefore undocumented.
+    /* DescriptorSetHostMappingFeaturesVALVE *//* VK_VALVE_descriptor_set_host_mapping */
+    //VkBool32           descriptorSetHostMapping;
+
+    // [DO NOT EXPOSE] its a D3D special use extension, shouldn't expose
+    /* MutableDescriptorTypeFeaturesVALVE *//* VK_VALVE_mutable_descriptor_type */
+    //VkBool32           mutableDescriptorType;
+
+    // [DO NOT EXPOSE]
+    /* SubpassShadingFeaturesHUAWEI *//* VK_HUAWEI_subpass_shading */
+    // VkBool32           subpassShading;
+
+    
+    /* Extensions Exposed as Features: */
+
+    /* VK_AMD_buffer_marker */
```

Comment:
might be useful for debugging later but requires a command buffer function and is definitely not supported in GL
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VK_AMD_buffer_marker.html

Reply:
> I say let's not expose unless we really feel the need for it

There's a `getNativeHandle()` in all our `IGPU...` objects, if the user wants to add an AMD specific path for the "facilitation of the development of debugging tools for tracking which pipelined command contributed to device loss", then let them.

Just don't add any extra methods to `ICommandBuffer`